jobs/component_promote.groovy

```diff
@@ -0,0 +1,60 @@
+evaluate(new File("${WORKSPACE}/common.groovy"))
+
+job('component-promote') {
+  description """
+    Promotes a component image to the production 'deis' registry org on e2e success after a merge to master
+  """.stripIndent().trim()
+
+
+  concurrentBuild()
+  throttleConcurrentBuilds {
+    maxPerNode(defaults.maxBuildsPerNode)
+    maxTotal(defaults.maxTotalConcurrentBuilds)
+  }
+
+  logRotator {
+    daysToKeep defaults.daysToKeep
+  }
+
+  publishers {
+    slackNotifications {
+      notifyFailure()
+      notifySuccess()
+    }
+  }
+
+  parameters {
+    stringParam('QUAY_USERNAME', 'deis+jenkins', 'Quay account name')
+    stringParam('QUAY_EMAIL', 'deis+jenkins@deis.com', 'Quay email address')
+    stringParam('COMPONENT_NAME', '', 'Component name')
+    stringParam('COMPONENT_SHA', '', 'Commit sha used for image tag')
+  }
+
+  wrappers {
+    buildName('${COMPONENT_NAME} promote #${BUILD_NUMBER}')
+    timestamps()
+    colorizeOutput 'xterm'
+    credentialsBinding {
+      string("QUAY_PASSWORD", "8317a529-10f7-40b5-abd4-a42f242f22f0")
+    }
+  }
+
+  steps {
+    shell '''
+      #!/usr/bin/env bash
+
+      set -eo pipefail
+
+      image_name_and_tag="${COMPONENT_NAME}:git-${COMPONENT_SHA:0:7}"
+      original_image=quay.io/deisci/"${image_name_and_tag}"
+      promoted_image=quay.io/deis/"${image_name_and_tag}"
+
+      echo "Promoting ${original_image} to ${promoted_image}"
+
+      docker login -e="\$QUAY_EMAIL" -u="\$QUAY_USERNAME" -p="\$QUAY_PASSWORD" quay.io
```

Comment:
@jchauncey I had discussed this with @mboersma offline and we came to the conclusion that just pushing to the quay registry was sufficient since `quay.io/deis/...` images are the ones listed in the release chart.  Meaning, once the release pipeline runs, it will look up the image promoted here (to quay), sub that in for the most recent semver release chart, run tests, and then, finally, retag with the official/'pretty' semver tag to both quay and dockerhub.

That being said, it would be easy to add dockerhub here, too.  I'm definitely open to that...